### PR TITLE
GSdx-d3d: Move Selector reset states in to a separate function.

### PR DESCRIPTION
**Pull request by lightningterror**
Also move vs_sel def in header file and rename it using the member name.

Code is easier to read/move/adjust similar to GL.

### CODE REVIEW

File: plugins/GSdx/GSRendererDX.cpp [hunk 1]
```diff
@@ -429,12 +435,10 @@ void GSRendererDX::DrawPrims(GSTexture* rt, GSTexture* ds, GSTextureCache::Sourc
...
 
-	GSDeviceDX::VSSelector vs_sel;
-
-	vs_sel.tme = PRIM->TME;
-	vs_sel.fst = PRIM->FST;
-	vs_sel.logz = dev->HasDepth32() ? 0 : m_logz ? 1 : 0;
-	vs_sel.rtcopy = !!rtcopy;
+	m_vs_sel.tme = PRIM->TME;
+	m_vs_sel.fst = PRIM->FST;
+	m_vs_sel.logz = dev->HasDepth32() ? 0 : m_logz ? 1 : 0;
```
Comment (ssakash) [inline plugins/GSdx/GSRendererDX.cpp]: Redundant ternary operator

File: plugins/GSdx/GSRendererDX.cpp [hunk 2]
```diff
@@ -429,12 +435,10 @@ void GSRendererDX::DrawPrims(GSTexture* rt, GSTexture* ds, GSTextureCache::Sourc
...
+	m_vs_sel.rtcopy = !!rtcopy;
```
Comment (ssakash) [inline plugins/GSdx/GSRendererDX.cpp]: I'm not sure if these sort of boolean type casts on a pointer variable is nice, maybe ``!= nullptr`` check would be much more readable? any opinion? (@refractionpcsx2  @turtleli )

Comment (turtleli) [reply plugins/GSdx/GSRendererDX.cpp]: I prefer `!= nullptr`.

File: plugins/GSdx/GSRendererDX.cpp [hunk 3]
```diff
@@ -429,12 +435,10 @@ void GSRendererDX::DrawPrims(GSTexture* rt, GSTexture* ds, GSTextureCache::Sourc
...
 
-	GSDeviceDX::VSSelector vs_sel;
-
-	vs_sel.tme = PRIM->TME;
-	vs_sel.fst = PRIM->FST;
-	vs_sel.logz = dev->HasDepth32() ? 0 : m_logz ? 1 : 0;
-	vs_sel.rtcopy = !!rtcopy;
+	m_vs_sel.tme = PRIM->TME;
+	m_vs_sel.fst = PRIM->FST;
+	m_vs_sel.logz = dev->HasDepth32() ? 0 : m_logz;
```
Comment (turtleli) [inline plugins/GSdx/GSRendererDX.cpp]: Maybe `!dev->HasDepth32() && m_logz`.

Comment (refractionpcsx2) [reply plugins/GSdx/GSRendererDX.cpp]: I already told him that, guess he didn't listen... :P

Comment (turtleli) [reply plugins/GSdx/GSRendererDX.cpp]: Message must have refracted and missed. :P

Comment (lightningterror) [reply plugins/GSdx/GSRendererDX.cpp]: Blame @ssakash he didn't like that change. I changed it to that at first :stuck_out_tongue: 

> ssakash: I don't know why you're going so deep into that logic while all I asked for is to remove the last ? 1 : 0 part which coverity will complain about

**Comment by lightningterror** [comment]
Updated with requested changes.

@turtleli can I merge this ?

File: plugins/GSdx/GSRendererDX.cpp [hunk 4]
```diff
@@ -429,12 +435,11 @@ void GSRendererDX::DrawPrims(GSTexture* rt, GSTexture* ds, GSTextureCache::Sourc
...
-
-	vs_sel.tme = PRIM->TME;
-	vs_sel.fst = PRIM->FST;
-	vs_sel.logz = dev->HasDepth32() ? 0 : m_logz ? 1 : 0;
-	vs_sel.rtcopy = !!rtcopy;
+	m_vs_sel.tme = PRIM->TME;
+	m_vs_sel.fst = PRIM->FST;
+	m_vs_sel.logz = !dev->HasDepth32() && m_logz;
+	if (rtcopy != nullptr)
+		m_vs_sel.rtcopy = !!rtcopy;
```
Comment (turtleli) [inline plugins/GSdx/GSRendererDX.cpp]: `m_vs_sel.rtcopy = rtcopy != nullptr;`

Comment (lightningterror) [reply plugins/GSdx/GSRendererDX.cpp]: Should be fine now.

**Comment by turtleli** [comment]
> can I merge this ?

It's probably fine?

**Pull request merged**